packages/js/components/src/form/form.tsx

```diff
@@ -61,6 +61,12 @@ type FormProps< Values > = {
 	validate?: ( values: Values ) => Record< string, string >;
 };
 
+function isChangeEvent< T >(
+	value: T | ChangeEvent< HTMLInputElement >
+): value is ChangeEvent< HTMLInputElement > {
+	return ( value as ChangeEvent< HTMLInputElement > ).target !== undefined;
+}
```

Comment:
Needed this as the `onChange` callback expected either a `string` or a `ChangeEvent` object.
Previously I was only expecting a `ChangeEvent` object, which didn't work when adding the Form to the product page.